test/gui/shared/scripts/bdd_hooks.py

```diff
@@ -141,3 +144,81 @@ def hook(context):
         raise Exception(
             "Step execution through test middleware failed. Error: " + e.read().decode()
         )
+
+
+def coredump(context):
+    cfg = ConfigParser()
+
+    # Select appropriate AUT binary with respect to the test run environment
+    try:
+        if environ.get('CI'):
+            cfg.read('/drone/src/test/gui/drone/server.ini')
+
+        else:
+            from pathlib import Path
+
+            HOME = str(Path.home())
+            cfg.read(HOME + '/.squish/ver1/server.ini')
+
+    except Exception as err:
+        print("the binary path can not be read")
+        print(err)
+
+    owncloudBinary = cfg.get('General', 'AUT/owncloud')
+    owncloudBinary = owncloudBinary.replace('"', '') + '/owncloud'
+
+    # read coredump location
+    coredumpFolderPath = subprocess.run(
+        ['cat', '/proc/sys/kernel/core_pattern'], stdout=subprocess.PIPE
+    ).stdout.decode('utf-8')
+    coredumpFolderPath = os.path.dirname(coredumpPath)
+    coredumpFilePattern = (
+        '^' + os.path.basename(coredumpPath).split('%')[0] + '.*owncloud$'
+    )
+
+    # list items from coredump path
+    ls = subprocess.run(['ls', '-t', coredumpFolderPath], capture_output=True)
+    # get coredump files only for owncloud desktop client
+    coredumps = (
+        subprocess.run(
+            ['grep', coredumpFilePattern],
+            input=ls.stdout,
+            stdout=subprocess.PIPE,
+        )
+        .stdout.decode('utf-8')
+        .split("\n")
+    )
+
+    if coredumps[len(coredumps) - 1] == "":
+        coredumps.pop()
+
+    if len(coredumps) == 1:
+        coredumpFilename = coredumps[0]
+        coredumpFile = os.path.join(coredumpFolderPath, coredumpFilename)
+        saveCoredump(owncloudBinary, coredumpFile, coredumpFilename)
+    elif len(coredumps) > 1:
```

Comment:
The `if` is superfluous. You can just loop over a list with a single entry as well.